jobs/future-effective-filings/k8s/templates/pipeline.json

```diff
@@ -0,0 +1,106 @@
+{
+    "kind": "Template",
+    "apiVersion": "v1",
+    "metadata": {
+        "name": "pipeline-template"
+    },
+    "objects": [
+        {
+            "kind": "BuildConfig",
+            "apiVersion": "v1",
+            "metadata": {
+                "name": "${NAME}-${PIPELINE_PURPOSE}-pipeline",
+                "creationTimestamp": null,
+                "labels": {
+                    "app": "${NAME}-${PIPELINE_PURPOSE}-pipeline",
+                    "name": "${NAME}-${PIPELINE_PURPOSE}-pipeline",
+                    "template": "${NAME}-${PIPELINE_PURPOSE}-pipeline"
+                }
+            },
+            "spec": {
+                "triggers": [
+                    {
+                        "type": "GitHub",
+                        "github": {
+                            "secretReference": {
+                                "name": "${WEBHOOK}"
+                            }
+                        }
+                    }
+                ],
+                "runPolicy": "Serial",
+                "source": {
+                    "type": "Git",
+                    "git": {
+                        "uri": "${GIT_REPO_URL}",
+                        "ref": "${GIT_REF}"
+                    },
+                    "contextDir": "${SOURCE_CONTEXT_DIR}"
+                },
+                "strategy": {
+                    "type": "JenkinsPipeline",
+                    "jenkinsPipelineStrategy": {
+                        "jenkinsfilePath": "${JENKINS_FILE}"
+                    }
+                },
+                "output": {},
+                "resources": {},
+                "postCommit": {},
+                "nodeSelector": {},
+                "successfulBuildsHistoryLimit": 5,
+                "failedBuildsHistoryLimit": 5
+            }
+        }
+    ],
+    "parameters": [
+        {
+            "name": "NAME",
+            "displayName": "Name",
+            "description": "The name assigned to all of the resources defined in this template.",
+            "required": true,
+            "value": "business-create-ui"
+        },
+        {
+            "name": "PIPELINE_PURPOSE",
+            "displayName": "Pipeline purpose",
+            "description": "The activity that this pipeline will manage. eg. build, test, promote, etc.",
+            "required": true,
+            "value": "build"
+        },
+        {
+            "name": "GIT_REPO_URL",
+            "displayName": "Git Repo URL",
+            "description": "The URL to your GIT repo.",
+            "required": true,
+            "value": "https://github.com/bcgov/bcrs-business-create-ui"
+        },
+        {
+            "name": "GIT_REF",
+            "displayName": "Git Reference",
+            "description": "The git reference or branch.",
+            "required": true,
+            "value": "master"
```

Comment:
I'm wondering if we should have special branch for testing so we can control what this picks up (ie, not necessarily the latest stuff committed to master). Thoughts?

Reply:
Regardless, this template should still be for master. This is just something that everyone can use off their local to use in an oc command so if we were going to change which branch we wanted the pipeline to build off then we would edit that for that one time. Also this only reflects the branch that the pipeline looks at for the jenkinsfile